### PR TITLE
EVEREST-633 Add multi-namespace support

### PR DESCRIPTION
**Pull request by recharte**
**Add multi-namespace support**
---
**Problem:**
EVEREST-633

Install everest and everest-operator into the `percona-everest` namespace and receive a list of namespaces to install DB operators to.
Each namespace gets an operator group with only its own namespace on it so that the DB operators access is limited to that namespace.
On the contrary, the `percona-everest` namespace gets an operator group with access to the entire list of namespaces in order to give the everest-operator permissions to manage all those namespaces. This list of namespaces is passed along to the everest-operator in the form of an environment variable called `WATCH_NAMESPACES`.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?

**Tests**
- [x] Is an Integration test/test case added for the new feature/change?
- [x] Are unit tests added where appropriate?

### CODE REVIEW

**Review by michal-kralik — APPROVED** [review]

**Review by oksana-grishchenko — COMMENTED** [review]
I'm a bit confused. The Epic is about the multi-k8s support, there is no information about the multi-namespaces support in Jira, is there any doc which describes how it's supposed to work? It's really hard to figure it out from the low level details even during the review and it's gonna be worse later in case this information was needed. 

Thank you for adding the PR description though. So the Everest operator is going to be installed into the `percona-everest` namespace and the operator can operate with the resources from the other namespaces, where the upstream operators and resources are gonna be installed. Am I understand it correctly that the Everest CRs (dbc, dbb, dbr etc) are gonna also be located in those specific namespaces, not int the `percona-everest`?

**Comment by recharte** [comment]
> I'm a bit confused. The Epic is about the multi-k8s support, there is no information about the multi-namespaces support in Jira, is there any doc which describes how it's supposed to work? It's really hard to figure it out from the low level details even during the review and it's gonna be worse later in case this information was needed.

Sorry about that, I haven't really updated any of the tickets/documentation that Andrew created a couple of months ago. The best we have is [this notion document](https://www.notion.so/percona/Multi-namespace-support-for-Everest-c2866849dbe64dd4acb5df98b5ff1660) that has a better description of the feature, even though I'm not sure it's fully up-to-date.

> Thank you for adding the PR description though. So the Everest operator is going to be installed into the `percona-everest` namespace and the operator can operate with the resources from the other namespaces, where the upstream operators and resources are gonna be installed. Am I understand it correctly that the Everest CRs (dbc, dbb, dbr etc) are gonna also be located in those specific namespaces, not int the `percona-everest`?

Yes, exactly. The  `percona-everest` will host the BackupStorages and MonitoringConfig CRs but all other CRs will only exist in their specific namespace where the DB operators are deployed to.

File: commands/install.go [hunk 1]
```diff
@@ -57,36 +58,22 @@ func newInstallCmd(l *zap.SugaredLogger) *cobra.Command {
...
 
 	cmd.Flags().Bool("operator.mongodb", true, "Install MongoDB operator")
 	cmd.Flags().Bool("operator.postgresql", true, "Install PostgreSQL operator")
 	cmd.Flags().Bool("operator.xtradb-cluster", true, "Install XtraDB Cluster operator")
-
-	cmd.Flags().String("channel.everest", "stable-v0", "Channel for Everest operator")
-	cmd.Flags().String("channel.victoria-metrics", "stable-v0", "Channel for VictoriaMetrics operator")
-	cmd.Flags().String("channel.xtradb-cluster", "stable-v1", "Channel for XtraDB Cluster operator")
-	cmd.Flags().String("channel.mongodb", "stable-v1", "Channel for MongoDB operator")
-	cmd.Flags().String("channel.postgresql", "fast-v2", "Channel for PostgreSQL operator")
```
[the remark this answers is not in the document]
Comment (recharte) [reply commands/install.go:72]: Exactly. We also want to manage the versions of the operators ourselves in the catalog releases so we don't need/want users to be messing with channels. For our current needs a single channel is enough.

File: cli-tests/tests/flow/all-operators.spec.ts [hunk 2]
```diff
@@ -81,59 +86,26 @@ test.describe('Everest CLI install', async () => {
...
-        'namespace: different-everest',
-      ]);
-      await cli.everestExec('uninstall --namespace=different-everest --assume-yes');
-      // Check that uninstall will fail because there's no everest deployment
-      out = await cli.everestExec('uninstall --namespace=different-everest --assume-yes');
-      await out.outErrContainsNormalizedMany([
-        'no Everest deployment in different-everest namespace',
-      ]);
-
-    });
```
[the remark this answers is not in the document]
Comment (recharte) [reply cli-tests/tests/flow/all-operators.spec.ts:110]: Right, the namespace where everest and everest-operator live in are hardcoded now so we don't have these use-cases anymore. From the product side we have an understanding that users can only deploy a single everest instance per k8s cluster so it's fine to hardcode this namespace.

**Review by oksana-grishchenko — APPROVED** [review]

**Pull request merged**